code/modules/mob/living/carbon/human/human_resist.dm

```diff
@@ -0,0 +1,112 @@
+/mob/living/carbon/human/process_resist()
+	//drop && roll
+	if(on_fire && !buckled)
+		adjust_fire_stacks(-1.2)
+		Weaken(3)
+		spin(32,2)
+		visible_message(
+			"<span class='danger'>[src] rolls on the floor, trying to put themselves out!</span>",
+			"<span class='notice'>You stop, drop, and roll!</span>"
+			)
+		sleep(30)
+		if(fire_stacks <= 0)
+			visible_message(
+				"<span class='danger'>[src] has successfully extinguished themselves!</span>",
+				"<span class='notice'>You extinguish yourself.</span>"
+				)
+			ExtinguishMob()
+		return TRUE
+
+	if(handcuffed)
+		spawn() escape_handcuffs()
+	else if(legcuffed)
+		spawn() escape_legcuffs()
+	else if(wear_suit && istype(wear_suit, /obj/item/clothing/suit/straight_jacket))
+		spawn() escape_straight_jacket()
+	else
+		..()
+
+/mob/living/carbon/human/proc/escape_straight_jacket()
+	setClickCooldown(100)
+
+	if(can_break_straight_jacket())
+		break_straight_jacket()
+		return
+
+	var/mob/living/carbon/human/H = src
+	var/obj/item/clothing/suit/straight_jacket/SJ = H.wear_suit
+
+	var/breakouttime = SJ.resist_time	// Configurable per-jacket!
+
+	var/attack_type = 0
+
+	if(H.gloves && istype(H.gloves,/obj/item/clothing/gloves/gauntlets/rig))
+		breakouttime /= 2	// Pneumatic force goes a long way.
+	else if(H.species.unarmed_types)
+		for(var/datum/unarmed_attack/U in H.species.unarmed_types)
+			if(istype(U, /datum/unarmed_attack/claws))
+				breakouttime /= 1.5
+				attack_type = 1
+				break
+			else if(istype(U, /datum/unarmed_attack/bite/sharp))
+				breakouttime /= 1.25
+				attack_type = 2
+				break
+
+	switch(attack_type)
```

Comment:
Magic numbers are bad :(
Some temporary ``#define``s would probably be better.
You can use ``#undef`` below the proc when done with them.